src/spineml/simulator/inputValue.cc

```diff
@@ -342,7 +350,21 @@ std::unique_ptr<SpineMLSimulator::InputValue::Base> SpineMLSimulator::InputValue
         return std::unique_ptr<Base>(new TimeVaryingArray(dt, numNeurons, node));
     }
     else if(strcmp(node.name(), "ExternalInput") == 0) {
-        return std::unique_ptr<Base>(new External(dt, numNeurons, node));
+        const std::string hostName = node.attribute("host").value();
+        if(hostName == "0.0.0.0") {
+            External *inputValue = new External(dt, numNeurons, node);
+
+            // Add to map of external inputs
+            const std::string name = node.attribute("name").value();
+            if(!externalInputs.emplace(name, inputValue).second) {
+                LOGW << "External input with duplicate name '" << name << "' encountered";
+            }
+
+            return std::unique_ptr<Base>(inputValue);
```

Comment:
Technically one could get a memory leak here if an exception is thrown in this block. If you make ``inputValue`` a ``unique_ptr`` from the outset you can avoid this and use ``get()`` to get the raw pointer out

Reply:
Very good point - fixed!